torch/quantization/fx/quantization_patterns.py

```diff
@@ -76,17 +77,33 @@ def __init__(self, node: Node, modules: Dict[str, torch.nn.Module]):
 
     def _maybe_get_last_node_only_observer(
         self,
-        modules: Dict[str, torch.nn.Module]
+        modules: Dict[str, torch.nn.Module],
+        equalization_enabled: bool = False,
     ) -> Optional[torch.nn.Module]:
         """
         If the last node of the pattern is observed, return the observer
         instance. Otherwise, return None.
         """
-        for maybe_obs_node, _ in self.last_node.users.items():
+        last_node = self.last_node
+
+        if equalization_enabled:
+            # Handle the additional mul node that gets added during equalization if
+            # it exists.
+            maybe_eq_node = maybe_get_next_module(last_node, modules, target_functional_type=torch.mul)
+            if maybe_eq_node is not None:
+                # Check that the mul node contains an 'equalization_scale' as an
+                # argument to ensure that this mul node is used for equalization.
+                for arg in maybe_eq_node.args:
+                    if isinstance(arg, Node) and "equalization_scale" in arg.name:
```

Comment:
actually Node name is a fragile thing to check I think, is it possible to check for something else?

Reply:
Could we check with node.target?